sdk/remoterendering/mixedreality-remoterendering/src/internal/renderingSession.ts

```diff
@@ -0,0 +1,202 @@
+   */
+  readonly teraflops?: number;
+  /**
+   * The time when the rendering session was created. Date and time in ISO 8601 format.
+   */
+  readonly createdOn?: Date;
+}
+
+/** The properties of a rendering session that may not be complete */
+/*
+export type PartialRenderingSessionProperties = {
+  [P in keyof RenderingSessionProperties]+?: RenderingSessionProperties[P];
+};
+*/
+
+/** The rendering session is ready for incoming connections. */
+export interface ReadyRenderingSession extends RenderingSessionBase {
+  /** The rendering session is ready for incoming connections. */
+  status: "Ready";
+  properties: RenderingSessionProperties;
+}
+
+/** The rendering session has encountered an error, and is unusable. */
+export interface ErrorRenderingSession extends RenderingSessionBase {
+  /** The rendering session has encountered an error, and is unusable. This is a terminal state. */
+  status: "Error";
+  /**
+   * The error object containing details about the rendering session startup failure.
+   */
+  readonly error: RemoteRenderingServiceError;
+  partialProperties: PartialRenderingSessionProperties;
+}
+
+/** The rendering session is starting, but not accepting incoming connections yet. */
+export interface StartingRenderingSession extends RenderingSessionBase {
+  /** The rendering session is starting, but not accepting incoming connections yet. */
+  status: "Starting";
+  partialProperties: PartialRenderingSessionProperties;
+}
+
+/** The rendering session enters the 'Expired' state when it has been in the 'Ready' state longer than its lease time. This is a terminal state. */
+export interface ExpiredRenderingSession extends RenderingSessionBase {
+  /** The rendering session enters the 'Expired' state when it has been in the 'Ready' state longer than its lease time. This is a terminal state. */
+  status: "Expired";
+  properties: RenderingSessionProperties;
+}
+
+/** The rendering session has been stopped with the 'Stop Session' operation. This is a terminal state. */
+export interface StoppedRenderingSession extends RenderingSessionBase {
+  /** The rendering session has been stopped with the 'Stop Session' operation. This is a terminal state. */
+  status: "Stopped";
+  partialProperties: PartialRenderingSessionProperties;
+}
+
+/** Information about a rendering session. This is a tagged union with "status" as its discriminant property. */
+export type RenderingSession =
+  | StartingRenderingSession
+  | ReadyRenderingSession
+  | ErrorRenderingSession
+  | ExpiredRenderingSession
+  | StoppedRenderingSession;
+
+function renderingSessionPropertiesFromSessionProperties(
+  session: SessionProperties
+): RenderingSessionProperties {
+  return {
+    arrInspectorPort: session.arrInspectorPort!,
+    handshakePort: session.handshakePort!,
+    elapsedTimeInMinutes: session.elapsedTimeInMinutes!,
+    host: session.host!,
+    teraflops: session.teraflops!,
+    createdOn: session.createdOn!
```

Comment:
It's worth checking them, although it would be an internal error if they are undefined, because the service is violating and _implicitly_ defined contract with the client.